src/coloring.rs

```diff
@@ -23,24 +23,40 @@ use rustworkx_core::coloring::{
     misra_gries_edge_color, two_color,
 };
 
+pub use rustworkx_core::coloring::GreedyStrategyCore;
+
+#[pyclass(module = "rustworkx")]
+#[derive(Clone, PartialEq)]
+pub enum GreedyStrategy {
+    Degree,
+    Saturation,
+}
+
```

Comment:
I need some help here. I want to define the greedy strategy enum inside rustworkx-core, and re-expose it here to the python interface. However, I wasn't able to get it to work, so I have created two different enums, one in rustworkx-core and one here. Is there a way to improve this?

Reply:
In addition, what do you think about the names "Degree", "Saturation", "IndependentSet"? Do you prefer instead "LargestFirst", "SaturationLargestFirst" or "IndependentSet", somewhat closer to networkx naming: https://networkx.org/documentation/stable/reference/algorithms/coloring.html.

Reply:
I can't think of a way to do it with one `enum`. The issue is we need to wrap an enum as a pyo3 pyclass to expose it in the rustworkx side to python. But we can't wrap the definition in rustworkx-core, so the only model that makes sense to me is having a duplicate pyclass version that we map to the rustworkx-core enum. The other option would be a string input I guess that we map to the enum, but that seems more error prone for this case.

Reply:
I'm fine with the naming you picked, but I think my initial thought would be to match the descriptions from `Adrian Kosowski, and Krzysztof Manuszewski, Classical Coloring of Graphs, Graph Colorings` which is what I think NetworkX used. Matching the terminology of the book and then citing it in the docstring gives people a good way to look up the exact coloring strategy if they so desire. But I like the more compact naming you used as it's easier to type. Maybe we should have a table at the end of the docstring that maps the strategy names to the book, I know you have it in the text written out, but a table for easy lookup might make it easier.

Reply:
What is the right way to add tables to rust documentation? I thought one should do this using pipes as explained in https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html, attempting this in [d227ad9](https://github.com/Qiskit/rustworkx/pull/1138/commits/d227ad9f0991c4fe91152457c2627e10b2554413), but this leads to error messages. Should this be done using `` .. list-table::`` or something like this?